app/src/main/kotlin/com/wire/android/ui/common/topappbar/CommonTopAppBarViewModel.kt

```diff
@@ -53,31 +54,42 @@ class CommonTopAppBarViewModel @Inject constructor(
     var state by mutableStateOf(CommonTopAppBarState())
         private set
 
-    private suspend fun currentScreenFlow() = currentScreenManager.observeCurrentScreen(viewModelScope)
+    private suspend fun currentScreenFlow() =
+        currentScreenManager.observeCurrentScreen(viewModelScope)
 
-    private fun connectivityFlow(userId: UserId): Flow<Connectivity> = coreLogic.sessionScope(userId) {
-        observeSyncState().map {
-            when (it) {
-                is SyncState.Failed, SyncState.Waiting -> Connectivity.WAITING_CONNECTION
-                SyncState.GatheringPendingEvents, SyncState.SlowSync -> Connectivity.CONNECTING
-                SyncState.Live -> Connectivity.CONNECTED
+    private fun connectivityFlow(userId: UserId): Flow<Connectivity> =
+        coreLogic.sessionScope(userId) {
+            observeSyncState().map {
+                when (it) {
+                    is SyncState.Failed, SyncState.Waiting -> Connectivity.WAITING_CONNECTION
+                    SyncState.GatheringPendingEvents, SyncState.SlowSync -> Connectivity.CONNECTING
+                    SyncState.Live -> Connectivity.CONNECTED
+                }
             }
         }
-    }
 
-    private suspend fun activeCallFlow(userId: UserId): Flow<Call?> = coreLogic.sessionScope(userId) {
-        calls.establishedCall().distinctUntilChanged().map { calls ->
-            calls.firstOrNull()
+    private suspend fun activeCallFlow(userId: UserId): Flow<Call?> =
+        coreLogic.sessionScope(userId) {
+            combine(
+                calls.establishedCall(),
+                calls.getIncomingCalls(),
+                calls.observeOutgoingCall(),
+            ) { establishedCall, incomingCalls, outgoingCalls ->
+                incomingCalls + outgoingCalls + establishedCall
+            }.map { calls ->
+                calls.firstOrNull()
+            }.distinctUntilChanged()
```

Comment:
This could be a bit weird...

What if I'm on a call, and some other conversation is calling now? Shouldn't we prioritise `established` calls first at the top app bar? So the user can more easily come back to the currently active call.

I think in this case it will prioritise `incoming`, then `outgoing`, then `established` last.

Might be as simple as reverting this. I'd also add tests to make sure that we don't break it.

Reply:
Indeed established call is not prioritised here, I'll change it and cover it with test